File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/attivita/dettagli/QuantitaSuImpianto.ts
import { Fabbricato } from 'app/Model/anagrafiche/Fabbricato';
import { EsercizioCDC } from '../centri_di_costo/EsercizioCDC';

export class QuantitaSuImpianto {
    Qta: number = 0;
    esercizioCDC: EsercizioCDC;
    Magazzino: Fabbricato;
    Lotto: string = '';

}

File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/attivita/dettagli/QuantitaSuImpianto.cs
using AgronicaCoreModelsSTD.anagrafiche;
using AgronicaCoreModelsSTD.attivita.centri_di_costo;
using AgronicaCoreModelsSTD.attivita.risorse;
using AgronicaCoreModelsSTD.metaschema.utilizzi;
using Newtonsoft.Json;
using System;

namespace AgronicaCoreModelsSTD.attivita.dettagli
{
    public class QuantitaSuImpianto
    {
        public decimal Qta { get; set; }
        public EsercizioCDC esercizioCDC { get; set; }
        public Fabbricato Magazzino { get; set; }
        public string Lotto { get; set; }
        public Prodotto Prodotto { get; set; }

        public QuantitaSuImpianto()
        {
            Lotto = "";
        }

        public QuantitaSuImpianto Clone()
        {
            try
            {
                string output = JsonConvert.SerializeObject(this);
                QuantitaSuImpianto deserializedObject = JsonConvert.DeserializeObject<QuantitaSuImpianto>(output);
                return deserializedObject;
            }
            catch (Exception ex)
            {
                throw new Exception("QuantitaSuImpianto.Clona: " + ex.Message);
            }
        }
    }
}
